pjmedia/src/pjmedia/vid_codec.c

```diff
@@ -218,6 +222,8 @@ PJ_DEF(pj_status_t) pjmedia_vid_codec_mgr_register_factory(
         pjmedia_vid_codec_info_to_id( &info[i],
                                   mgr->codec_desc[mgr->codec_cnt+i].id,
                                   sizeof(pjmedia_codec_id));
+        pj_strdup2_with_null(mgr->pool, &mgr->codec_list[mgr->codec_cnt+i],
```

Comment:
No need to duplicate? Just point to `mgr->codec_desc[mgr->codec_cnt+i].id`.
This way, - we don't need pool, and - later during factory unregistration, we can also erase the codec string from the list.

Reply:
I just realise that if we don't duplicate the string, the list obtained by sdp_neg can point to an invalid string if the codec has been unregistered. Unregistering a codec should be quite an uncommon use case, but still, I guess we have no choice but use the pool here.